File: test_cors.js
fetch("https://tradelense-ai-engine.fly.dev/analyze", {
  method: "OPTIONS",
  headers: {
    "Origin": "https://traderlense.com",
    "Access-Control-Request-Method": "POST",
    "Access-Control-Request-Headers": "content-type"
  }
}).then(async res => {
  console.log("OPTIONS Status:", res.status);
  res.headers.forEach((v, k) => console.log(k, ":", v));

  if (res.ok) {
    console.log("\nStarting POST...");
    const postRes = await fetch("https://tradelense-ai-engine.fly.dev/analyze", {
      method: "POST",
      headers: {
        "Origin": "https://traderlense.com",
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        asset: "BTC/USD - Time Horizon: SCALP - Bias: LONG - Rationale: Trendline bounce"
      })
    });
    console.log("POST Status:", postRes.status);
    postRes.headers.forEach((v, k) => console.log(k, ":", v));
    const text = await postRes.text();
    console.log("Response Body:", text);
  }
}).catch(err => console.error("Error:", err));

File: test_cors.py
import requests
import json

headers = {
    'Origin': 'https://traderlense.com',
    'Access-Control-Request-Method': 'POST',
    'Access-Control-Request-Headers': 'content-type'
}
res = requests.options('https://tradelense-ai-engine.fly.dev/analyze', headers=headers)
print("OPTIONS Headers:", res.headers)
print("OPTIONS Status:", res.status_code)

post_headers = {
    'Origin': 'https://traderlense.com',
    'Content-Type': 'application/json'
}
data = {"asset": "BTC/USD - Time Horizon: SCALP - Bias: LONG - Rationale: Trendline bounce"}
try:
    print("Sending POST request...")
    res2 = requests.post('https://tradelense-ai-engine.fly.dev/analyze', headers=post_headers, json=data)
    print("POST Status:", res2.status_code)
    print("POST Response:", res2.text)
    print("POST Headers:", res2.headers)
except Exception as e:
    print("Error:", e)
